vmaas-go/base/utils/gin.go

```diff
@@ -105,6 +106,25 @@ func respStatusError(c *gin.Context, code int, err error) {
 	})
 }
 
+func processBadRequestErrMessage(err error) error {
+	errMessage := err.Error()
+	if strings.HasPrefix(errMessage, "parsing time") {
+		parts := strings.Split(errMessage, `"`)
+		if len(parts) < 2 {
+			return errors.New("Wrong date format (not ISO format with timezone)")
+		}
+		return errors.New("Wrong date format (not ISO format with timezone): " + parts[1])
+	}
+	if strings.HasSuffix(errMessage, "looking for beginning of value") {
+		parts := strings.Split(errMessage, ` `)
+		if len(parts) < 3 {
+			return errors.New("malformed input")
+		}
+		return errors.New("malformed input: invalid character " + parts[2])
+	}
+	return err
+}
+
```

Comment:
Parsing errors based on error message and then changing the error message is not optimal. I'd prefer wrapping original error.
Error for wrong date can be for example handled when UnmarshalJSON for time fails (it would require custom type for time so UnmarshalJSON can be defined, something like https://github.com/RedHatInsights/patchman-engine/blob/master/base/types/timestamp.go). Anyways, timestamp is parsed only by `bindValidateJSON` so at least move the current logic to that function.

`looking for beginning of value` is returned by regex error? In that case you can wrap the error in vmaas-lib before returning regex error or return some custom error type and then change the message after checking the error type with errors.Is

Reply:
The original error `parsing time "2017-01-01T00:00:00" as "2006-01-02T15:04:05Z07:00": cannot parse "" as "Z07:00"` is clunky and requires some go-specific knowledge to understand. That's why I opted to replace it rather than to just wrap it.

`invalid character 'N' looking for beginning of value` also occurs during JSON parsing, for example, when there is `NaN` instead of a number.

I will move the logic. Should I also wrap the errors instead?